src/coreclr/jit/optimizer.cpp

```diff
@@ -7608,17 +7608,805 @@ bool Compiler::optIsRangeCheckRemovable(GenTree* tree)
+
+    if (optReturnBlock)
+    {
+        // Get rid of the third block
+        comp->fgUnlinkBlock(b3);
+        b3->bbFlags |= BBF_REMOVED;
+        // If b3 was the last block of a try or handler, update the EH table.
+        comp->ehUpdateForDeletedBlock(b3);
+    }
+
+    if (!optReturnBlock)
+    {
+        // Update bbRefs and bbPreds
+        //
+        // Replace pred 'b2' for 'b2->bbNext' with 'b1'
+        // Remove  pred 'b2' for 'b2->bbJumpDest'
+        comp->fgReplacePred(b2->bbNext, b2, b1);
+        comp->fgRemoveRefPred(b2->bbJumpDest, b2);
+    }
+
+    // Update loop table
+    comp->fgUpdateLoopsAfterCompacting(b1, b2);
+    if (optReturnBlock)
+    {
+        comp->fgUpdateLoopsAfterCompacting(b1, b3);
+    }
+}
+
+//
+//  optOptimizeBoolsReturnBlock: Optimize boolean when b1.bbJumpKind is BBJ_COND and b2.bbJumpKind is BBJ_RETURN
+//
+//  Arguments:
+//      comp    The instance of Compiler class
+//
+//  Returns:
+//      changeReturn    Set to true if boolean optimization is done and b1, b2 and b3 are folded into b1.
+//
+//  Notes:
+//      Below members of OptBoolsDsc are set when called:
+//          b1      The first Basic Block with the BBJ_COND conditional jump type
+//          b2      The next basic block of b1 with the BBJ_RETURN
+//          b3      The next basic block of b2 with GT_RETURN
+//
+//      For example, (x==0 && y==0) generates:
+//          b1: GT_JTRUE (BBJ_COND)
+//          b2: GT_RETURN (BBJ_RETURN)
+//          b3: GT_RETURN (BBJ_RETURN),
+//      and it is folded into
+//          b1: GT_RETURN (BBJ_RETURN)
+//
+//      if b1.bbJumpDest == b2.bbNext, it transforms
+//          b1 : brtrue(t1, b3)
+//          b2 : ret(t2)
+//          b3 : ret(0)
+//      to
+//          b1 : ret((!t1) && t2)
+//
+bool OptBoolsDsc::optOptimizeBoolsReturnBlock(Compiler* comp)
+{
+    assert(b1 != nullptr && b2 != nullptr && b3 != nullptr);
+
+    bool changeReturn = false;
+
+    sameTarget          = false;
+    Statement* const s1 = optOptimizeBoolsChkBlkCond();
+    if (s1 == nullptr)
+    {
+        return changeReturn;
+    }
+
+    // Find the branch conditions of b1 and b2
+
+    GenTree* const op1 = optIsBoolComp(comp, &t1);
+    if (op1 == nullptr)
+    {
+        return changeReturn;
+    }
+    c1 = op1;
+
+    GenTree* const op2 = optIsBoolComp(comp, &t2);
+    if (op2 == nullptr)
+    {
+        return changeReturn;
+    }
+    c2 = op2;
+
+    // Find the type and cost conditions of t1 and t2
+
+    if (!optOptimizeBoolsChkTypeCostCond(comp))
+    {
+        return changeReturn;
+    }
+
+    // Get the fold operator and the comparison operator
+
+    var_types foldType = c1->TypeGet();
+    if (varTypeIsGC(foldType))
+    {
+        foldType = TYP_I_IMPL;
+    }
+    this->foldType = foldType;
+
+    foldOp = GT_NONE;
+    cmpOp  = GT_NONE;
+
+    optReturnGetFoldAndCompOper();
+    if (foldOp == GT_NONE || cmpOp == GT_NONE)
+    {
+        return changeReturn;
+    }
```

Comment:
```suggestion
    if (!optReturnGetFoldAndCompOper())
    {
        return false;
    }
```
and make `optReturnGetFoldAndCompOper()` return true on success, false on failure.

Reply:
Done.
Per previoud comment, removed the function `optReturnGetFoldAndCompOper()` and moved the code into this method because it was called only from this method.